tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -319,6 +319,28 @@ def test_tune_num_leaves(self):
             assert runner.lgbm_params['num_leaves'] != unexpected_value
             assert len(tuning_history) == 20
 
+    def test_tune_num_leaves_negative_max_depth(self):
+        # type: () -> None
+
+        params = {
+            'metric': 'binary_logloss',
+            'max_depth': -1,
+        }  # type: Dict[str, Any]
+        X_trn = np.random.uniform(10, size=50).reshape((10, 5))
```

Comment:
[nit] We can omit `reshape` here.

```suggestion
        X_trn = np.random.uniform(10, size=(10, 5))
```

Reply:
Thanks for the suggestion! Comitted.